cmd/ipfs/pinmfs.go

```diff
@@ -185,7 +189,10 @@ func pinMFS(
 		}
 		return lastPin{}, err
 	}
-	if pinFound {
+
+	// CID of the current MFS root is already pinned, nothing to do
+	if alreadyPinned {
+		log.Infof("pinning MFS to %s: pin for %s already exists, skipping", svcName, cid)
 		return lastPin{}, nil
 	}
```

Comment:
(This is a bug in the original PR too.) We should return here something like
lastPin{Time: time.Now(), ServiceName: svcName, ServiceConfig: svcConfig, CID: cid},
because upstream lastPin{} is taken to mean failed pinning.